sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/kafka/KafkaTestTable.java

```diff
@@ -163,8 +159,12 @@ public synchronized Map<TopicPartition, OffsetAndTimestamp> offsetsForTimes(
           }
         };
 
-    for (String topic : getTopics()) {
-      consumer.updatePartitions(topic, partitionInfoMap.get(topic));
+    for (Map.Entry<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> entry :
+        kafkaRecords.entrySet()) {
+      consumer.updatePartitions(
```

Comment:
won't this update the partitions multiple times for each partition?

Reply:
Good point, I'll move that to a separate loop:
```
for (Map.entry<String, PartitionInfo> entry : partitionInfoMap.entrySet()) {
  consumer.updatePartitions(entry.getKey(), entry.getValue());
}
```